File: spec/java/src/io/kaitai/struct/spec/TestImportsAbsRel.java
package io.kaitai.struct.spec;

import io.kaitai.struct.testformats.ImportsAbsRel;
import io.kaitai.struct.testformats.ImportedAndRel;
import io.kaitai.struct.testformats.ImportedRoot;
import org.testng.annotations.Test;
import static org.testng.Assert.*;
public class TestImportsAbsRel extends CommonSpec {

    @Test
    public void testImportsAbsRel() throws Exception {
        ImportsAbsRel r = ImportsAbsRel.fromFile(SRC_DIR + "fixed_struct.bin");

        assertIntEquals(r.one(), 80);
        assertIntEquals(r.two().one(), 65);
        assertIntEquals(r.two().two().one(), 67);
    }
}

File: spec/rust/tests/test_imports_abs_rel.rs
use std::fs;

extern crate kaitai;
use self::kaitai::*;
use rust::formats::imports_abs_rel::*;

#[test]
fn test_imports_abs_rel() {
    let bytes = fs::read("../../src/fixed_struct.bin").unwrap();
    let _io = BytesReader::from(bytes);
    let res = ImportsAbsRel::read_into(&_io, None, None);
    let r : OptRc<ImportsAbsRel>;

    if let Err(err) = res {
        panic!("{:?}", err);
    } else {
        r = res.unwrap();
    }

    assert_eq!(*r.one(), 80);
    assert_eq!(*r.two().one(), 65);
    assert_eq!(*r.two().two().one(), 67);
}
